src/adapter/src/catalog/builtin.rs

```diff
@@ -2964,6 +2956,47 @@ IN CLUSTER mz_introspection
 ON mz_catalog.mz_secrets (schema_id)",
 };
 
+pub const MZ_CLUSTERS_IND: BuiltinIndex = BuiltinIndex {
+    name: "mz_clusters_ind",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE INDEX mz_clusters_ind
+IN CLUSTER mz_introspection
+ON mz_catalog.mz_clusters (id, name)",
+};
+
+pub const MZ_CLUSTER_REPLICAS_IND: BuiltinIndex = BuiltinIndex {
+    name: "mz_cluster_replicas_ind",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE INDEX mz_cluster_replicas_ind
+IN CLUSTER mz_introspection
+ON mz_catalog.mz_cluster_replicas (id, cluster_id, name, size, availability_zone)",
+};
+
+pub const MZ_CLUSTER_REPLICA_UTILIZATION_IND: BuiltinIndex = BuiltinIndex {
+    name: "mz_cluster_replica_utilization_ind",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql:
+        "CREATE INDEX mz_cluster_replica_utilization_ind
+IN CLUSTER mz_introspection
+ON mz_internal.mz_cluster_replica_utilization (replica_id, process_id, cpu_percent, memory_percent)",
```

Comment:
What is the purpose of changing the key to have multiple columns?

Reply:
Consider:

```
materialize=> create table t (x int, y int, z int);
CREATE TABLE
materialize=> create index i on t (x);
CREATE INDEX
materialize=> explain select * from t where x = 5;
                       Optimized Plan                        
-------------------------------------------------------------
 Explained Query (fast path):                               +
   Project (#0..=#2)                                        +
     ReadExistingIndex materialize.public.i lookup_value=(5)+
                                                            +
 Used Indexes:                                              +
   - materialize.public.i                                   +
 
(1 row)

materialize=> drop index i;
DROP INDEX
materialize=> create index i on t (x,y,z);
CREATE INDEX
materialize=> explain select * from t where x = 5;
               Optimized Plan               
--------------------------------------------
 Explained Query (fast path):              +
   Filter (#0 = 5)                         +
     ReadExistingIndex materialize.public.i+
                                           +
 Used Indexes:                             +
   - materialize.public.i                  +
 
(1 row)

```

Note that in the first example, we can look up "x=5" directly in the index, whereas in the second example we have to read everything, because `x` is not the key of the index (rather, `(x,y,z)` is).

Reply:
Ah thanks for the explanation, I assumed we could still do lookups even with extra keys

Reply:
We discussed this on Slack, but for posterity: the syntax `CREATE INDEX i ON t (x, y, z)` does _not_ create an index with three keys (there is no such concept in Materialize). All indexes have _one_ key, and this syntax causes the key to have three columns. Thus, you need to have all three columns in order to look something up in the index and avoid a full scan.